.github/workflows/reusable_provenance.yaml

```diff
@@ -166,24 +92,40 @@ jobs:
           api_key = '${{ secrets.ENT_API_KEY }}'
           EOF
 
-      - name: Upload to Ent (post-merge only)
-        if: github.event_name == 'push'
-        id: ent_upload
+      - name: Download the built artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ needs.generate_provenance.outputs.build-outputs-name }}
+          path: downloads
+
+      - name: Download the DSSE document
+        uses: actions/download-artifact@v3
+        with:
+          name:
+            ${{ needs.generate_provenance.outputs.attestations-download-name }}
+          path: downloads
+
+      - name: Upload binary to Ent
+        id: ent_upload_binary
+        working-directory: downloads
         run: |
-          echo "provenance_digest=$(ent put ${{ needs.build_binary.outputs.attestation-name }})" >> $GITHUB_OUTPUT
+          echo "binary_digest=$(ent put ${{ needs.get_inputs.outputs.artifact-path }})" >> $GITHUB_OUTPUT
 
-      # Add the artifact digest and name to the ./comment file.
-      - name: Format artifact digest
+      - name: Upload provenance to Ent
+        id: ent_upload_provenance
+        working-directory: downloads
         run: |
-          echo -e "Artifact digest and name:\n\n\`\`\`" >> ./comment
-          echo "${{ needs.build_binary.outputs.digest }}" | base64 -d | tee -a ./comment
-          echo -e "\`\`\`" >> ./comment
+          echo "provenance_digest=$(ent put ${{ needs.get_inputs.outputs.provenance-name }})" >> $GITHUB_OUTPUT
 
       # Add the provenance digest to the ./comment file.
-      - name: Format provenance digest
+      - name: Format artifact and provenance digest (post-merge only)
         run: |
+          echo -e "Artifact name: $(basename ${{ needs.get_inputs.outputs.artifact-path }})" >> ./comment
+          echo -e "Artifact digest:\n\n\`\`\`" >> ./comment
+          echo ${{ steps.ent_upload_binary.outputs.binary_digest }} | tee -a ./comment
+          echo -e "\`\`\`" >> ./comment
           echo -e "\n\nProvenance digest:\n\n\`\`\`" >> ./comment
-          echo ${{ steps.ent_upload.outputs.provenance_digest }} | tee -a ./comment
+          echo ${{ steps.ent_upload_provenance.outputs.provenance_digest }} | tee -a ./comment
```

Comment:
One thought I had (and it's really beyond the scope of this PR): would it possible to add the provenance digests as tags to the PR?

Reply:
Do you mean git tags? Or GitHub labels?

Reply:
GitHub labels, yes.